### DISCUSSION TITLE
Checkbox attribute

### DISCUSSION THREAD

**Discussion by musumbidenis** [Power-Components/livewire-powergrid#645] [Q&A] [answered]
I have noticed that PowerGrid's uses the "id" attribute to fetch selected rows. How can I change that to my table's primary key such that I can export data for selected rows only

**Comment by dansysanalyst** [comment, marked as answer]
Hi @musumbidenis,

You must specify the primary key, read more [here](https://livewire-powergrid.com/#/table/datasource?id=custom-primary-key) and set your "id" in the [showCheckBox](https://livewire-powergrid.com/#/table/features-setup?id=showcheckbox) method.

The documentation needs some clarification on this.

For example, I have created a database table `restaurants` with the id `restaurant_id`.

```php
        Schema::create('restaurants', function (Blueprint $table) {
            $table->id('restaurant_id');
            $table->string('name');
            $table->timestamps();
        });
```

This is how my PowerGrid table looks like:

```php 
<?php

final class RestaurantTable extends PowerGridComponent
{
    use ActionButton;

    public string $primaryKey = 'restaurants.restaurant_id';
    public string $sortField = 'restaurants.restaurant_id';

    public function setUp(): array
    {
        $this->showCheckBox('restaurant_id');

//....
```

**Reply by musumbidenis** [reply]
Thank you so much it worked fine
